### PR TITLE
PYTHON-2774 Migrate snappy testing to from Ubuntu 18 to amazon1-2018

### PR DESCRIPTION
**Pull request by ShaneHarvey**

### CODE REVIEW

**Comment by ShaneHarvey** [comment]
Testing that this works now that libsnappy-dev has been installed.

**Comment by ShaneHarvey** [comment]
This is working:
```
 [2021/07/12 23:23:12.774] Looking in indexes: https://artifactory.corp.mongodb.com/artifactory/api/pypi/pypi/simple, https://pypi.org/simple
 [2021/07/12 23:23:13.075] Collecting python-snappy==0.5.1
 [2021/07/12 23:23:13.075]   Downloading https://artifactory.corp.mongodb.com/artifactory/api/pypi/pypi/packages/4c/8f/09f0a11dbcaedf9b0758a37ab2bb77d6c34b9c29afc4d0440019c152d2af/python-snappy-0.5.1.tar.gz (15 kB)
 [2021/07/12 23:23:13.573] Building wheels for collected packages: python-snappy
 [2021/07/12 23:23:13.573]   Building wheel for python-snappy (setup.py): started
 [2021/07/12 23:23:17.247]   Building wheel for python-snappy (setup.py): finished with status 'done'
 [2021/07/12 23:23:17.247]   Created wheel for python-snappy: filename=python_snappy-0.5.1-cp27-cp27mu-linux_x86_64.whl size=27129 sha256=02839744fe0beef1eaa31fd664ccbd4e79d1207a021dfd46ba995def4aa88ae8
 [2021/07/12 23:23:17.247]   Stored in directory: /home/ec2-user/.cache/pip/wheels/da/54/d7/61b4efe553be3f397b1352755890e2dc63982492dcfdbbca41
 [2021/07/12 23:23:17.837] Successfully built python-snappy
 [2021/07/12 23:23:17.837] Installing collected packages: python-snappy
 [2021/07/12 23:23:17.837] Successfully installed python-snappy-0.5.1
 [2021/07/12 23:23:18.492] Running noauth tests over nossl with python python
 [2021/07/12 23:23:18.492] 2.7.18 (default, Jun  8 2021, 18:24:43)
 [2021/07/12 23:23:18.492] [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
 ```

File: .evergreen/config.yml [hunk 1]
```diff
@@ -2443,12 +2428,12 @@ buildvariants:
   rules:
     # Server versions 3.6+ support snappy and zlib.
     - if:
+        platform: "*"
         python-version: "*"
         c-extensions: "*"
         compression: ["snappy", "zlib"]
       then:
         add_tasks:
-          - "test-5.0-standalone"
```
Comment (prashantmital) [inline .evergreen/config.yml:2451]: Why get rid of 5.0 testing?

Comment (ShaneHarvey) [reply .evergreen/config.yml:2451]: We still test against 5.0 the "test-5.0-standalone" here was redundant.

Comment (prashantmital) [reply .evergreen/config.yml:2451]: Ah, got it.

**Review by prashantmital — APPROVED** [review]

**Pull request merged**